packages/shared/src/components/onboarding/FilterOnboarding.tsx

```diff
@@ -35,6 +37,13 @@ export function FilterOnboarding({
     }
   };
 
+  useEffect(() => {
+    const hasSelectedTopics = Object.values(selectedTopics).some(
+      (value) => value === true,
+    );
+    if (setHasCategories) setHasCategories(hasSelectedTopics);
+  }, [selectedTopics, setHasCategories]);
+
```

Comment:
Since `setHasCategories` is optional we can early return before ever calculating `hasSelectedTopics`.

Reply:
Also, after looking at the code of `onChangeSelectedTopic` I think we don't have to add another callback and just base it on the `onSelectedTopics` callback. This would also reduce the usage of `useEffect`.